src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -334,17 +332,10 @@ class Peer(
         }
         val session = LightningSession(enc, dec, ck)
 
-        suspend fun send(message: ByteArray) {
-            try {
-                session.send(message) { data, flush -> socket.send(data, flush) }
-            } catch (ex: TcpSocket.IOException) {
-                logger.warning { "TCP send: ${ex.message}" }
-                closeSocket(ex)
-            }
-        }
-
-        logger.info { "sending init $ourInit" }
-        send(LightningMessage.encode(ourInit))
+        // TODO use atomic counter instead
```

Comment:
There doesn't seem to be a portable implementation `AtomicInt` unfortunately. I can fallback to `UUID`...

Reply:
This seems to be the library for multiplatform atomic operations, but I'm not sure if it's worth importing right now: https://github.com/Kotlin/kotlinx-atomicfu

Reply:
Yeah that seems a bit risky. I think we can stick with milliseconds. Maybe `UUID` to be safe?

Reply:
I'm not sure what is best between UUID and timestamp...we shouldn't have two connections established at the same millisecond, so timestamp is probably fine?